contentcuration/contentcuration/view/node_views.py

```diff
@@ -278,3 +359,157 @@ def _move_node(node, parent=None, sort_order=None, channel_id=None):
                 n.tags.add(t)
 
     return node
+
+def sync_nodes(request):
+    logging.debug("Entering the sync_nodes endpoint")
+
+    if request.method != 'POST':
+        return HttpResponseBadRequest("Only POST requests are allowed on this endpoint.")
+    else:
+        data = json.loads(request.body)
+
+        try:
+            nodes = data["nodes"]
+            channel_id = data['channel_id']
+
+        except KeyError:
+            return ObjectDoesNotExist("Missing attribute from data: {}".format(data))
+
+        all_nodes = []
+        with transaction.atomic():
```

Comment:
I've been seeing the `with transaction.atomic():...with delay_mptt_updates` in a couple of places. Would it be good to wrap these inside a single context manager, for brevity?